File: jdroid-java-webapp-sample/src/main/java/com/jdroid/javaweb/sample/config/ConfigFirebaseRepository.java
package com.jdroid.javaweb.sample.config;

import com.jdroid.java.firebase.database.PairFirebaseRepository;
import com.jdroid.java.firebase.database.auth.CustomTokenFirebaseAuthenticationStrategy;
import com.jdroid.java.firebase.database.auth.FirebaseAuthenticationStrategy;
import com.jdroid.javaweb.sample.context.ServerApplication;

public class ConfigFirebaseRepository extends PairFirebaseRepository {

	@Override
	protected FirebaseAuthenticationStrategy createFirebaseAuthenticationStrategy() {
		return new CustomTokenFirebaseAuthenticationStrategy() {
			@Override
			protected String getAuthToken() {
				return ServerApplication.get().getFirebaseAuthToken();
			}
		};
	}

	@Override
	protected String getFirebaseUrl() {
		return ServerApplication.get().getFirebaseUrl();
	}

	@Override
	protected String getPath() {
		return "config";
	}
}

File: jdroid-java-webapp-sample/src/main/java/com/jdroid/javaweb/sample/config/ConfigFirebaseRepository.kt
package com.jdroid.javaweb.sample.config

import com.jdroid.java.firebase.database.PairFirebaseRepository
import com.jdroid.java.firebase.database.auth.CustomTokenFirebaseAuthenticationStrategy
import com.jdroid.java.firebase.database.auth.FirebaseAuthenticationStrategy
import com.jdroid.javaweb.sample.context.ServerApplication

class ConfigFirebaseRepository : PairFirebaseRepository() {

    override val firebaseUrl: String = ServerApplication.get().firebaseUrl

    override val path: String = "config"

    override fun createFirebaseAuthenticationStrategy(): FirebaseAuthenticationStrategy? {
        return object : CustomTokenFirebaseAuthenticationStrategy() {
            override val authToken: String
                get() = ServerApplication.get().firebaseAuthToken
        }
    }
}
